modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpCache.java

```diff
@@ -56,18 +62,27 @@ <T extends AbstractResponse> T putIfAbsent(
     ) {
         // can't use cache.computeIfAbsent due to the elevated permissions for the jackson (run via the cache loader)
         CacheKey cacheKey = new CacheKey(ip, databasePath);
+        long cacheStart = System.nanoTime();
         // intentionally non-locking for simplicity...it's OK if we re-put the same key/value in the cache during a race condition.
         AbstractResponse response = cache.get(cacheKey);
+        long cacheRequestTime = System.nanoTime() - cacheStart;
 
         // populate the cache for this key, if necessary
         if (response == null) {
+            long start = System.nanoTime();
             response = retrieveFunction.apply(ip);
+            long databaseRequestTime = System.nanoTime() - start;
+            storeQueryTimeInNanos.addAndGet(databaseRequestTime);
+            missesTimeInNanos.addAndGet(cacheRequestTime);
+            missesTimeInNanos.addAndGet(databaseRequestTime);
```

Comment:
It's the tiniest thing, but how about adding these longs together before calling `addAndGet` (once)?